### DISCUSSION TITLE
Horizontal Bar Chart

### DISCUSSION THREAD

**Discussion by watuLammini** [patrykandpatrick/vico#379] [Ideas]
Hi,

first, thanks for providing this nice chart library!
I have created a bar chart in my app using your library in compose, but now it would be super helpful if I could have a horizontal version of this bar chart (meaning the bars going sideways instead of upwards). Especially for charts with a lot of bars, this uses up the screen space much better, when the device is in portrait orientation (and I don't want to force my users to rotate their devices for just this one screen).
Maybe I didn't get it, but there does not seem to be any possibility for it using this library. And obviously I don't want to switch libraries and start all over again...

So maybe other people have similar demands and you could implement this feature? Otherwise any other ideas and workarounds are of course welcome.

**Comment by patrickmichalik** [comment]
Hello! Unfortunately, there’s no such feature at the moment, but we’ll be looking into adding horizontal bar charts down the road.

**Reply by binshadpix** [reply]
Hi, is the horizontal bar chart added?

**Reply by Gowsky** [reply]
Hello, horizontal bar charts have not been added yet.

**Reply by mohamedCrowdAnalyzer** [reply]
> Hello! Unfortunately, there’s no such feature at the moment, but we’ll be looking into adding horizontal bar charts down the road.

Are there any updates regarding this feature?

**Reply by Gowsky** [reply]
Hello, @mohamedCrowdAnalyzer. When we have an update, we'll share it here. Please note that you can always refer to the documentation to find out what chart types are available.

**Comment by binshadpix** [comment]
https://github.com/patrykandpatrick/vico/discussions/967

hi patrick,
it would be great if you answer this concern

**Reply by Gowsky** [reply]
Hello, @binshadpix. Please stay on topic of the discussion. We're notified about every new discussion, and we respond as soon as we can.

**Comment by marcosalis** [comment]
Hi, I'm finding myself in need of a horizontal bar chart as well.
It's a pity this isn't available as I've seen so far that all elements of a chart could be rotated. I was playing with "manually" rotating the entire chart composable with `Modifier.rotate(90f)` and disabling horizontal scrolling, but I'm running on some (understandable) issues with the composable sizing (the composable's height, once rotated, isn't correct).
@Gowsky would you think of a workaround that could allow this without any library modifications? Thank you!

**Reply by Gowsky** [reply]
Hello, @marcosalis. Can you be more specific about the issue? You can correct the width and height by using `Modifier.width` and `Modifier.height` keeping in mind they are inverted. To achieve an equivalent of `Modifier.fillMaxWidth`, you may have to check the available width and explicitly pass that value to `Modifier.height`.

**Reply by marcosalis** [reply]
Hi @Gowsky, thank you for your reply! I did try that, unfortunately the Chart composable doesn't seem to size itself correctly even if I specify a fixed height/width. I'll post an example with code to reproduce as soon as I manage.
However, the `rotate` option isn't a full substitute to supporting horizontal bar charts because the vertical scrolling (which I will require) won't work anyway.

**Reply by Gowsky** [reply]
> However, the rotate option isn't a full substitute to supporting horizontal bar charts

Certainly not — it's a workaround, as you've mentioned, and one that we wouldn't recommend. As per @patrickmichalik's comment, we plan on adding horizontal bar charts in the future.

> However, the `rotate` option isn't a full substitute to supporting horizontal bar charts because the vertical scrolling (which I will require) won't work anyway.

It seems to work for the most part, but flinging appears to be broken. You could resolve this by disabling scrolling via `rememberVicoScrollState`, adding your own gesture detection at the right point in the modifier chain, and calling `VicoScrollState.scroll` there. When `VicoScrollState.scrollEnabled` is `false`, the default zoom settings change so that the content fills the viewport, but you can adjust this via `rememberVicoZoomState`.

**Reply by marcosalis** [reply]
Thank you for your input @Gowsky, I did try a few workarounds, but the result is quite sketchy, and full customisation would be impossible without dedicating it a lot of time. I will have to fall back to another library for the time being for horizontal charts.
Looking forward for the feature to be implemented on Vico's side, obviously!

**Reply by PepaZapletal** [reply]
@marcosalis Hi, can you please share your solution? What library did you use? Thank you for the help.

**Reply by marcosalis** [reply]
I think I'll be going with MPAndroidChart for the time being. It's far from ideal as I only needed to flip the existing chart, but the needed decorations won't work with Vico on horizontal charts.

**Comment by PepaZapletal** [comment]
Hi, 

can I ask if there is an estimated date when this feature will be available? 

So it is currently not possible to easily create this type of chart right?

![Snímek obrazovky 2025-04-09 v 11 42 13](https://github.com/user-attachments/assets/e2135a48-8f53-4e1d-bf41-ac33e9abf375)

**Reply by Gowsky** [reply]
Hello, @PepaZapletal. This is a FOSS that we work on in our free time, thus we don't provide any ETAs.
> So it is currently not possible to easily create this type of chart right?

That's correct, unless you use some workarounds. But we don't recommend them.
